Update service defaults and add filtering details (#3138)
dotnet/docs-aspire: docs/fundamentals/service-defaults.md

- Maps the health checks endpoint to `/health`.
- Maps the liveness endpoint to `/alive` route where the health check tag contains `live`.

> [!NOTE]

> > Starting with .NET Aspire 9.2, the [starter template](aspire-sdk-templates.md#solution-templates) has been updated to include a call to <xref:Aspire.Hosting.ResourceBuilderExtensions.WithHttpsHealthCheck*> for ASP.NET Core projects. To prevent request traces for the configured health endpoints (`/health` and `/alive`) from cluttering the request trace logs in the dashboard during development, the service defaults templates now filter out these traces by default.

Reviewer: This sounded a little odd to me and I can't tell if there is a single change mentioned here with two effects or two distinct changes? Here's an idea, but I don't know if it's accurate.

```suggestion
> Starting with .NET Aspire 9.2, the [starter template](aspire-sdk-templates.md#solution-templates) has been updated to include a call to <xref:Aspire.Hosting.ResourceBuilderExtensions.WithHttpsHealthCheck*> for ASP.NET Core projects. This change prevents request traces for the configured health endpoints (`/health` and `/alive`) from cluttering the request trace logs in the dashboard during development. The template's default values for the service filter out these traces.
```

Reviewer 2: @adegeo I think your proposal changes the meaning to the point it's not factual 😄 

The change to call `WithHttpHealthCheck` in the app host project doesn't prevent the traces, it causes them. It's the subsequent changes in the service defaults project that filters out those traces.

> > Starting with .NET Aspire 9.2, the [starter template](aspire-sdk-templates.md#solution-templates) has been updated to include a call to <xref:Aspire.Hosting.ResourceBuilderExtensions.WithHttpsHealthCheck*> for ASP.NET Core projects. To prevent request traces for the configured health endpoints (`/health` and `/alive`) from cluttering the request trace logs in the dashboard during development, the service defaults templates now filter out these traces by default.

Author: ```suggestion
> Starting with .NET Aspire 9.2, the [starter template](aspire-sdk-templates.md#solution-templates) has been updated to include a call to <xref:Aspire.Hosting.ResourceBuilderExtensions.WithHttpsHealthCheck*> for ASP.NET Core projects. To keep the request trace logs cleaner during development, traces for the configured health endpoints (`/health` and `/alive`) are now excluded by default in the **Service Defaults** project template.
```
